### ISSUE TITLE
No git gutter indicators are showing

### ISSUE THREAD

**Issue by ludvigeriksson** [microsoft/vscode#60010]
- VSCode Version: 1.27.2
- OS Version: 10.13.6

Neither the file explorer nor the gutter shows changes in my git project. If I click the source control tab I can see the changes and view them side by side. But I want to see changes in the file explorer and gutters as explained here: https://code.visualstudio.com/docs/editor/versioncontrol#_gutter-indicators

Steps to Reproduce:

1. Run VS code
2. Open git project
3. Modify something

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by joaomoreno** [comment]
Is is a theme issue perhaps? Does it reproduce with another theme?

Can you show me your full settings file?

**Comment by rightaway** [comment]
I'm also getting this issue but only since upgrading to 1.28.0. I tried changing the color theme and it doesn't help.

**Comment by birros** [comment]
I have exactly the same problem as @ludvigeriksson. I did the test using a new installation of VSCode from the standalone version, so the configuration file is the default file, no extension is installed.

The result is the same with the standalone version used on Fedora and the Flatpak version (installed from Flathub) also used on Fedora. I don't think the problem is related to the OS. 

I haven't found a method to debug VSCode to find the cause. **How to debug VSCode in this case?**

```
Version: 1.28.1
Commit: 3368db6750222d319c851f6d90eb619d886e08f5
Date: 2018-10-11T18:09:20.566Z
Electron: 2.0.9
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

OS: Fedora Silverblue
Version: 29.20181014.n.0
Commit: cfecaa3a580a8e506f9e94f7ec3d81ddbdb223fe1fd6af77f67bc0cf0fe00d78
Date: 2018-10-14 11:45:17
Architecture: x86_64
```

**Comment by birros** [comment]
@joaomoreno The Source Control panel works but does not automatically detect changes and requires manual refresh. I can still display the diffs and make commits.

The status bar displays the current branch and I can switch to another branch.

**Comment by xoxota99** [comment]
Yes, everything works, but no color coding in Explorer panel, and no change indicators in editor gutters.

**Comment by joaomoreno** [comment]
@xoxota99 @birros Try running with `code --verbose` and see if file events are running.

@ludvigeriksson Any updates?

**Comment by ludvigeriksson** [comment]
I’ve noticed that it isn’t permanent, it occasionally works. As others have said, the source control panel works as expected, but the color coding of files and in the gutter disappears sometimes. Unfortunately I’m out of town right now and can’t run it verbose, but I can try it when I get back next week.

**Comment by birros** [comment]
No events from the File Watcher.

```
[main 9:57:44 PM] Starting VS Code
[main 9:57:44 PM] from: /var/home/julien/VSCode-linux-x64/resources/app
[main 9:57:44 PM] args: { _: [],
  help: false,
  h: false,
  version: false,
  v: false,
  wait: false,
  w: false,
  diff: false,
  d: false,
  add: false,
  a: false,
  goto: false,
  g: false,
  'new-window': false,
  n: false,
  'unity-launch': false,
  'reuse-window': false,
  r: false,
  'open-url': false,
  performance: false,
  p: false,
  'prof-startup': false,
  verbose: true,
  logExtensionHostCommunication: false,
  'disable-extensions': false,
  disableExtensions: false,
  'list-extensions': false,
  'show-versions': false,
  nolazy: false,
  issue: false,
  'skip-getting-started': false,
  'skip-release-notes': false,
  'sticky-quickopen': false,
  'disable-restore-windows': false,
  'disable-telemetry': false,
  'disable-updates': false,
  'disable-crash-reporter': false,
  'skip-add-to-recently-opened': false,
  status: false,
  s: false,
  'file-write': false,
  'file-chmod': false,
  'driver-verbose': false,
  trace: false }
[main 9:57:44 PM] Resolving machine identifier...
[main 9:57:44 PM] Resolved machine identifier: 301fab43cfc53a35e8ab69c908b1207d9be05ff7d9925e59f4ed7cb8af05fd41
[main 9:57:44 PM] update#setState idle
[main 9:57:44 PM] windowsManager#open
[main 9:57:44 PM] telemetry/optInStatus { optIn: true }
bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
[main 9:57:45 PM] windowsService#whenSharedProcessReady
[main 9:57:45 PM] windowsService#getRecentlyOpened 1
[main 9:57:45 PM] windowsService#setRepresentedFilename 1
[main 9:57:45 PM] windowsService#startCrashReporter
[main 9:57:45 PM] windowsService#getRecentlyOpened 1
[main 9:57:46 PM] IPC#vscode-workbenchLoaded
[main 9:57:46 PM] windowsService#setRepresentedFilename 1
[main 9:57:46 PM] windowsService#getWindowCount
[main 9:57:46 PM] RequestService#request https://vscode-update.azurewebsites.net/api/update/linux-x64/stable/3368db6750222d319c851f6d90eb619d886e08f5
[main 9:57:46 PM] IPC#vscode:broadcast vscode:changeColorTheme {"baseTheme":"vs-dark","background":"#1e1e1e"}
[main 9:57:46 PM] RequestService#request https://vscode-update.azurewebsites.net/api/update/linux-x64/stable/3368db6750222d319c851f6d90eb619d886e08f5
[main 9:57:46 PM] windowsService#isFocused 1
[main 9:57:47 PM] windowsService#getWindowCount
[main 9:58:01 PM] Lifecycle#window-before-close 1
[main 9:58:01 PM] Lifecycle#unload() 1
[main 9:58:01 PM] Lifecycle#unload(): unload continues without veto 1
[main 9:58:01 PM] Lifecycle#onBeforeWindowClose.fire()
[main 9:58:01 PM] Lifecycle#window-before-close 1
[main 9:58:01 PM] Lifecycle#window-close 1
[main 9:58:01 PM] Lifecycle#window-closed 1
[main 9:58:01 PM] Lifecycle#onShutdown.fire()
[main 9:58:01 PM] Lifecycle#window-all-closed
[main 9:58:01 PM] Lifecycle#before-quit
[main 9:58:01 PM] Lifecycle#onBeforeShutdown.fire()
[main 9:58:01 PM] App#will-quit: disposing resources
```

**Comment by xoxota99** [comment]
Not sure where to look for "events", but here's the output from stdout when I run in Verbose mode:

> /Applications/Visual Studio Code.app ./Contents/Resources/app/bin/code --verbose
[main 1:16:57 PM] Starting VS Code
[main 1:16:57 PM] from: /Applications/Visual Studio Code.app/Contents/Resources/app
[main 1:16:57 PM] args: { _: [],
  help: false,
  h: false,
  version: false,
  v: false,
  wait: false,
  w: false,
  diff: false,
  d: false,
  add: false,
  a: false,
  goto: false,
  g: false,
  'new-window': false,
  n: false,
  'unity-launch': false,
  'reuse-window': false,
  r: false,
  'open-url': false,
  performance: false,
  p: false,
  'prof-startup': false,
  verbose: true,
  logExtensionHostCommunication: false,
  'disable-extensions': false,
  disableExtensions: false,
  'list-extensions': false,
  'show-versions': false,
  nolazy: false,
  issue: false,
  'skip-getting-started': false,
  'skip-release-notes': false,
  'sticky-quickopen': false,
  'disable-restore-windows': false,
  'disable-telemetry': false,
  'disable-updates': false,
  'disable-crash-reporter': false,
  'skip-add-to-recently-opened': false,
  status: false,
  s: false,
  'file-write': false,
  'file-chmod': false,
  'driver-verbose': false,
  trace: false }
[main 1:16:57 PM] Resolving machine identifier...
[main 1:16:57 PM] Resolved machine identifier: 8b93e61cbb3025cc072ef63fcdb4ce963ba59f2141f0c62a26981496fc16508a
[main 1:16:57 PM] update#setState idle
[main 1:16:57 PM] windowsManager#open
2018-10-20 13:16:57.315 Electron[5643:162145] *** WARNING: Textured window <AtomNSWindow: 0x7fc2be479b60> is getting an implicitly transparent titlebar. This will break when linking against newer SDKs. Use NSWindow's -titlebarAppearsTransparent=YES instead.
[main 1:16:57 PM] telemetry/optInStatus { optIn: true }
[5643:1020/131658.006499:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.018476:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.024881:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.091073:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[main 1:16:58 PM] IPC#vscode-workbenchLoaded
[main 1:16:58 PM] windowsService#whenSharedProcessReady
[main 1:16:58 PM] windowsService#getRecentlyOpened 1
[main 1:16:58 PM] windowsService#getWindowCount
[5643:1020/131658.246884:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.247107:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.247302:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.247590:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[main 1:16:58 PM] windowsService#startCrashReporter
[5643:1020/131658.285216:INFO:CONSOLE(1409)] "%c INFO", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1409)
[main 1:16:58 PM] windowsService#getRecentlyOpened 1
[5643:1020/131658.309202:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.375677:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[main 1:16:58 PM] RequestService#request https://vscode-update.azurewebsites.net/api/update/darwin/stable/7f3ce96ff4729c91352ae6def877e59c561f4850
[5643:1020/131658.435506:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[main 1:16:58 PM] IPC#vscode:broadcast vscode:changeColorTheme {"baseTheme":"vs-dark","background":"#1e1e1e"}
[main 1:16:58 PM] windowsService#updateTouchBar 1
[main 1:16:58 PM] windowsService#isFocused 1
[main 1:16:58 PM] windowsService#getWindowCount
[5643:1020/131658.496817:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.547826:INFO:CONSOLE(282)] "%c[IPC Library: File Watcher (chokidar)] %cStart watching: /Users/myusername/Google Drive/projects/RPI/Cubey]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (282)
[5643:1020/131658.547889:INFO:CONSOLE(282)] "%c[IPC Library: File Watcher (chokidar)] %cWatcher basePath does not match version on disk and was corrected (original: /Users/myusername/Google Drive/projects/RPI/Cubey, real: /Users/myusername/Google Drive/projects/RPI/cubey)", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (282)
[5643:1020/131658.574236:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.576217:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.585156:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.588625:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.589180:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.589623:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.742838:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.928472:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.928758:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.929043:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.929382:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.929583:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.929836:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.930070:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131658.979521:INFO:CONSOLE(282)] "%c[Extension Host] %cactivating extension %c(at Object.activate (/Users/myusername/.vscode/extensions/ms-vscode.cpptools-0.19.0/out/src/LanguageServer/extension.js:41:13))", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (282)
[5643:1020/131659.332565:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131659.335845:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131659.378243:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131659.379572:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131659.384931:INFO:CONSOLE(282)] "%c[Extension Host] %crunCommand", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (282)
[5643:1020/131659.426384:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131659.632924:INFO:CONSOLE(282)] "%c[Extension Host] %cFound PIO Core 3.6.1-rc.4 %c(at t.<anonymous> (/Users/myusername/.vscode/extensions/platformio.platformio-ide-1.4.1/node_modules/platformio-node-helpers/lib/index.js:1:43161))", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (282)
[5643:1020/131659.637024:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131659.638733:INFO:CONSOLE(282)] "%c[Extension Host] %crunCommand", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (282)
[5643:1020/131701.112210:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131701.112435:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[5643:1020/131708.081750:INFO:CONSOLE(1408)] "%cTRACE", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (1408)
[main 1:17:27 PM] update#checkForUpdates, state =  idle
[main 1:17:27 PM] update#setState checking for updates
[main 1:17:27 PM] telemetry/update:notAvailable { explicit: false }
[main 1:17:27 PM] update#setState idle

**Comment by xoxota99** [comment]
Other types of color coding in explorer (such as files containing errors) are showing up fine.

**Comment by joaomoreno** [comment]
@xoxota99 When running in verbose, make file changes in your workspace outside of VS Code, does VS Code react to those file events? Do the file events show up in the logs?

Does this feature actually work sometimes and then it stops working after a while?

**Comment by sharm294** [comment]
I'm having the same issue. The indicators were working file but I upgraded to 1.28.2 (on Ubuntu 16.04) and now it doesn't seem to be working. I've tried after disabling all the extensions as well. Unfortunately, I don't remember the past version but it was at least 1.27 I think.

@joaomoreno When I edit a file outside of Code when it's running with --verbose, I do see events occur on the terminal showing that the file was changed and it updates in the VS Code window as well but still no git gutter indicators.

**Comment by joaomoreno** [comment]
@sharm294 Does it repro on every repository?

**Comment by ludvigeriksson** [comment]
It might be worth noting that I also have the project open in a git-client [(git-fork)](https://git-fork.com/), which does fetch the remote periodically. However, I have not made any changes/commits in either code or git-fork all day, neither before nor after the gutter stopped working.

**Comment by joaomoreno** [comment]
> It happens for me on multiple different repos

The question is rather: does it happen on **all** repos, or just on a subset? Because if it's just a subset, please find out what is common between them.

**Comment by Argysh** [comment]
My company pushed version 1.28.2 on our workstations (Oracle Linux Server 7.5) today and since then the gutter icons are gone in **all** of my repositories with or without any extensions enabled.

**Comment by joaomoreno** [comment]
What about your coworkers, did the same issue hit them?

**Comment by xoxota99** [comment]
> @xoxota99 When running in verbose, make file changes in your workspace outside of VS Code, does VS Code react to those file events? Do the file events show up in the logs?
> 
> Does this feature actually work sometimes and then it stops working after a while?

Switching to the SCM panel, all changes are showing up normally. Git itself works fine. Integration appears to work correctly across the board. Only thing missing is the gutter change indicators, and the color coding int he explorer panel.

This issue is showing up on one computer, but not on another. Both are Macbooks, both have the same OS version, same VSCode version, and same plugins installed.

I'm not currently on the computer where this is happening, so can't verify if this is happening across all repos or not.

**Comment by xoxota99** [comment]
CONFIRMED: This is not occurring across all repos. Only the current one that I'm working on. Hopefully this narrows it down a bit. What can I provide in terms of differences between the repos that might help?

**Comment by sharm294** [comment]
I've noticed something that's interesting. My repos are accessible through a symlink. If I open the repo folder in Code through the symlink path, then the gutter indicators don't work (on the left by the line nor on the overview on the right). In the Explorer window, I don't see that M indicating the file has been modified.  However, both of those things work if I open the folder through the symlink free path. Same behavior if I open a single file through a symlinked path vs a non-symlinked path.

**Comment by joaomoreno** [comment]
> What can I provide in terms of differences between the repos that might help?

Filesystem location, unicode characters in path, anything of special interest...

**Comment by birros** [comment]
I just tested @sharm294's suggestion, I confirm this behavior. Under Fedora Silverblue `/home -> /var/home`, so no repo in the user folder opened with VSCode allows to benefit from Gutter.

**Comment by sharm294** [comment]
@joaomoreno I didn't realize this was an issue. My repo paths haven't changed but the indicators were definitely working before I updated. From that link, it looks like the issue was addressed for non-root symlinks? That's my case.

It is still being recognized as a Git repo even through a symlink (i.e. the source control button on the left show all the pending changes) and that was the same behavior before as well.

**Comment by xoxota99** [comment]
> > What can I provide in terms of differences between the repos that might help?
> 
> Filesystem location, unicode characters in path, anything of special interest...

Filesystem location: /Users/myusername/Google\ Drive/projects/RPI/cubey

Things of special interest: This is in my Google Drive folder, so it is synced across two different macbooks, via Google Sync. One one computer, the color coding works fine. On the other, it doesn't. As far as I know, there are no symlinks, Unicode characters, or other shenanigans in any of the file paths (other than the space between "Google" and "Drive"). If it helps, I can zip up the entier thing and send it to you. It's also available from Github [here](https://github.com/xoxota99/cubey).

**Comment by FyFou** [comment]
Same issue for me. I don't get it, it worked fine before vacation and now i don't have the gutter indicator. Even on older repo, it just doesn't work for me everywhere.

**Comment by FyFou** [comment]
I don't know why it came back in between two save.

**Comment by xoxota99** [comment]
Unfortunately, in the interim, my laptop crapped out and I had to reimage it. The issue is no longer showing up on my new laptop, though nothing about this filesystem path has changed (everything was restored from backup).

**Comment by planetrocky** [comment]
Same issue here, and has been for many months now.  Zero gutter markings of any kind visible..

For me, this is a critical feature, being able to see my recent changes.  My productivity has declined since this bug/problem was introduced.

**Comment by joaomoreno** [comment]
If someone can get me a 100% reproducible case, I'll fix it on the day.

It seems @planetrocky reproduces it 100% of the time. Would you be willing to do a screen sharing session with em?

**Comment by brennenberkley** [comment]
I just wanted to add that I'm also having this issue with some workspaces, and in those workspaces it applies to all of the project folders. None of my projects or workspaces use symlinks or weird names. I opened the same file in 3 separate VS Code windows: one with just the project, one in workspace A, and one in workspace B. The git colors in the gutter show up when I'm in the project folder and workspace A, but not in workspace B. The highlighting in the file tree works fine in all 3 places.

I'm running version 1.29.1 on macOS Mojave

**Comment by joaomoreno** [comment]
@brennenberkley Can you give me exact path names, so I can reproduce on my side?

**Comment by brennenberkley** [comment]
@joaomoreno I recreated all of my workspaces and that fixed the issue, but I've done that before and it came back eventually in the new workspaces. But all of the projects are in `/Users/Brennen/work/`, and most of them are rails projects. One of the workspaces that was giving me trouble was `/Users/Brennen/VS Code Workspaces/ALL.code-workspace` where the contents of that file look like this: (I can't figure out how to insert multi-line code blocks, sorry)
`{
	"folders": [
		{
			"path": "/Users/brennen/work/acts-as-taggable-on"
		},
		{
			"path": "/Users/brennen/work/bing-api"
		},
		{
			"path": "/Users/brennen/work/chimera"
		},
	]
}`

**Comment by joaomoreno** [comment]
@brennenberkley But you can no longer reproduce?

Blocks of code are possible with triple-backticks.

**Comment by ababak** [comment]
I have the same problem.

It appears that git gutter does not work when the project file is opened as "~/Documents..." but does work when the file is open by right-clicking in SCM view as now it has an absolute path: "/mnt/c/Users/..."

Hope this helps.

**Comment by ababak** [comment]
Further investigation shows that I have a symlink from "~/Documents" to "/mnt/c/Users/...", maybe this is the reason?

**Comment by avalsoff** [comment]
I have the same issue after last update. GIt diff indicators aren't showing.

**Comment by SynderBlack** [comment]
I can reproduce this, no gutter icons and no "M" icon on modified files (in the explorer view, the M appears on the SCM view) when using a link. Everything works properly when the actual path is opened instead.

**Comment by logTom** [comment]
I can also reproduce this issue when I open a folder inside a symlink to a different hard drive.

**Comment by tommyvct** [comment]
Yes, avoid symbol link did the trick.

**Comment by jaimeagudo** [comment]
I can reproduce https://github.com/Microsoft/vscode/issues/60010#issuecomment-464920512 On Mac OS Mojave 18C54 with vscode Version 1.31.1 (1.31.1), no NTFS at all

**Comment by chris-miaskowski** [comment]
OS: Ubuntu
VSCode: 1.32.0
Configuration: my _development_ folder is symlinked to an NTFS partition

I can also confirm that removing the symlink **solved the problem**.

Interestingly it was fine until recently (not sure when it broke exactly, but ~ month ago?)

**Comment by BrandonJamesBorders** [comment]
Does this make any sesnse? When I open VS Code from Github Desktop (Ctrl + Shift + A) I don't get file tree git colors. If I open the folder from Windows Explorer (right click and select Open with VS Code) I get file tree git colors and the M tag on modified files.

**Comment by dark-swordsman** [comment]
```
Version: 1.32.3
Commit: a3db5be9b5c6ba46bb7555ec5d60178ecc2eaae4
Date: 2019-03-14T23:38:49.842Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Linux x64 4.15.0-46-generic
```

I just started having this issue as well, however, I noticed it only start happening with a recent project.

On all my previous projects, there was only one package.json in the highest directory. or two package.jsons in two different folders in one workspace. In this project I have two package.jsons in two subdirectories of one folder. Also, I believe this only started happening after the second package.json was added.

I'm on a plane right now, so I'm not gonna try to do any debugging, but I'll try to do some extra testing tomorrow or the next day.

**Comment by jansim** [comment]
For me, this issue occurs reliably after I switch to a directory via the Project Manager extension `alefragnani.project-manager`. It does not occur, when I open a directory directly in vscode by hand.

- Version 1.32.3 (1.32.3)
- MacOS 10.14.3
- Project Manager: 10.4.0 (2019-03-14)

**Comment by magpie514** [comment]
I am getting this issue as well. Repository contains no symlinks, it resides in an ext4 partition, and gutter indicators will not show. However, closing the project and reopening the folder makes them show again, at the expense of losing the current workbench layout.

**Comment by darkvertex** [comment]
I can reliably get non-updating git gutters in Linux if symlinks are involved.

At my employer someone made the odd call to symlink `/home` to a different mount path, so everything under `~/` or `/home/alan` for me is technically symlinked and I don't have enough admin rights to right this wrong.

Anyway, when I have a workspace with paths like `/home/alan/repos/blablabla` the git gutters do not seem to refresh automatically, only after I browse into the git panel and let it discover the state of the repo there, then it works for the current state of the file, but not interactively if I go away from the git repo pane and keep working on the file.

My theory is that whatever subsystem watches for filesystem IO events is receiving events for the "true" mount paths and thus not correlating said events to "opened files" in the editor that appear under _a different path_ (due to their symlinked nature.) The correlation is thus only made when the git pane is involved and the repo is actively refreshed (as opposed to autorefreshed by events like editing a file after freshly opening Code.)

Once I realized this I worked around it by setting my workspace folder paths to their "true" explicit mount paths (not under /home but under /Volumes/blablabla...) and then my git gutters continued refreshing perfectly without hassle.

It would be nice to investigate this bug in the interest of other confused users who suffer the same fate as me.

---

Info from Help->About:

* Version: 1.34.0
* Commit: a622c65b2c713c890fcf4fbf07cf34049d5fe758
* Date: 2019-05-15T21:55:35.507Z
* Electron: 3.1.8
* Chrome: 66.0.3359.181
* Node.js: 10.2.0
* V8: 6.6.346.32
* OS: Linux x64 3.10.0-693.11.6.el7.x86_64

**Comment by codycraven** [comment]
I was encountering this issue and discovered it was because the folder I opened in VS Code was a symlink. When I opened the folder directly my gutter indicators appeared properly.

**Comment by darkvertex** [comment]
@joaomoreno Do you feel like the symlink gutter issue (now confirmed by @codycraven just like it happened to me) is worth a new ticket or do we keep the conversation going in here? Please advice.

Thank you.

**Comment by cosmolux** [comment]
Confirmed.  I can open the same file in two windows, one via a symlink, and the symlink gutter shows no indicator, while the other does.

**Comment by jblachly** [comment]
I can confirm that opening the same folder with/without navigating through symlink reproduces the problem exactly.

Fresh install of vscode on Ubuntu18 VM (connecting remotely from Mac)

Directories when accessed through `~/source/` do not show gutter marks (nor does the "modified" dot in the directory view sidebar update).

Directories when accessed through `/mnt/scratch/source` **DO** show gutter marks and does update the "modified" dot in the directory view sidebar.

**Comment by joaomoreno** [comment]
There's already a symlink issue: https://github.com/microsoft/vscode/issues/5970

And I just pushed a fix for: https://github.com/microsoft/vscode/issues/41085 (try it tomorrow on insiders)

---

So, if you are not using symlinks and you are not opening a subfolder of the repo, and you still don't see any automatic updating on the viewlet, let me know.

**Comment by bmakan** [comment]
I face a similar issue. Project structure:
`/src`
`/submodule` (git submodule)
`/src/symlink -> ../submodule`

When I open the `/src` folder I cannot see the git change gutters made in the `submodule`'s files in the editor.

I also use remote development, but I opened the same project directly on the server and had the same issue there, so I can confirm it is not caused by remote development.

**Comment by jansim** [comment]
Just for reference, I was able to figure out what the problem was for me (since it was not resolved in v1.36.1):

I was switching directories via an extension (`alefragnani.project-manager`) but I had originally provided a base folder with an uppercase letter instead of lowercase e.g. `/path/to/Lorem/` vs `path/to/lorem/`. This was automatically handled by vscode and would only trigger a warning in the console (`Watcher basePath does not match version on disk and was corrected`), but gutter indicators would stop working. I was able to fix this by using the proper upper/lower case letters in the base folder.

I don't know if this really is a proper bug that needs to be handled on VSCode's end, since the problem was on my end in providing a wrong directory path. It is kind of unexpected, though, that vscode will happily open the directory, but then not display any gutters.

**Comment by MHM5000** [comment]
Maybe my situation will help:
A Parent git without submodules but there are 2 folders named `client` and `server` which are 2 separate git repos. 2 repos are not committed to the parent one!

```
Ubuntu 18.04
VsCode:
1.37.1
f06011ac164ae4dc8e753a3fe7f9549844d15e35
x64
```

**Comment by Mochnant** [comment]
I had this same issue, and was able to figure out that it was related to a path vs repo casing disconnect.  Somehow the repo `WorkTools` has been put in a directory `/worktools/`.  When I changed it to `/WorkTools/` the gutter indicators began appearing.
